src/main/java/com/google/api/codegen/DocConfig.java

```diff
@@ -0,0 +1,130 @@
+  public abstract String getApiName();
+
+  public abstract String getMethodName();
+
+  public abstract String getReturnType();
+
+  public abstract InitCode getInitCode();
+
+  public abstract ImmutableList<InputParameter> getParams();
+
+  /**
+   * DocConfig builder minimum functionality
+   */
+  public abstract static class Builder<BuilderType extends Builder<BuilderType>> {
+    private static final String REQUEST_PARAM_DOC =
+        "The request object containing all of the parameters for the API call.";
+
+    private static final String REQUEST_PARAM_NAME = "request";
+
+    protected abstract BuilderType setInitCodeProxy(InitCode initCode);
+
+    protected abstract BuilderType setParamsProxy(ImmutableList<InputParameter> params);
+
+    @SuppressWarnings("unchecked")
+    public BuilderType setRequestObjectInitCode(
+        GapicContext context, Interface service, Method method) {
+      Map<String, Object> initFieldStructure = createInitFieldStructure(context, service, method);
+      InitCodeGenerator generator = new InitCodeGenerator();
+      InitCode initCode = generator.generateRequestObjectInitCode(method, initFieldStructure);
+      setInitCodeProxy(initCode);
+      return (BuilderType) this;
+    }
+
+    @SuppressWarnings("unchecked")
+    public BuilderType setFieldInitCode(
+        GapicContext context, Interface service, Method method, Iterable<Field> fields) {
+      Map<String, Object> initFieldStructure = createInitFieldStructure(context, service, method);
+      InitCodeGenerator generator = new InitCodeGenerator();
+      InitCode initCode =
+          generator.generateRequestFieldInitCode(method, initFieldStructure, fields);
+      setInitCodeProxy(initCode);
+      return (BuilderType) this;
+    }
+
+    private static Map<String, Object> createInitFieldStructure(
+        GapicContext context, Interface service, Method method) {
+      MethodConfig methodConfig =
+          context.getApiConfig().getInterfaceConfig(service).getMethodConfig(method);
+      Map<String, String> fieldNamePatterns = methodConfig.getFieldNamePatterns();
+
+      ImmutableMap.Builder<String, InitValueConfig> initValueConfigMap = ImmutableMap.builder();
+      for (Map.Entry<String, String> fieldNamePattern : fieldNamePatterns.entrySet()) {
+        CollectionConfig collectionConfig =
+            context.getCollectionConfig(service, fieldNamePattern.getValue());
+        InitValueConfig initValueConfig =
+            InitValueConfig.create(context.getApiWrapperName(service), collectionConfig);
+        initValueConfigMap.put(fieldNamePattern.getKey(), initValueConfig);
+      }
+      Map<String, Object> initFieldStructure =
+          FieldStructureParser.parseFields(
+              methodConfig.getSampleCodeInitFields(), initValueConfigMap.build());
+      return initFieldStructure;
+    }
+
+    public BuilderType setRequestObjectParam(
+        @SuppressWarnings("unused") GapicContext context,
```

Comment:
Good point. I removed them.

I guess thought certain language might need them in this base class definition, but they can add them back if needed in the future.